### ISSUE TITLE
Use `no_grad()` in generate

### ISSUE THREAD

**Issue by hash-ir** [hash-ir/nanoGPT#1]
Memory consumption of [`generate`](https://github.com/hash-ir/transformer-lab/blob/a15c8bc8dae9292147e4cd3e3b07be2b0ebd4dda/models.py#L78) can be reduced by using `torch.no_grad()` context manager. Also, this will be required, for example when dropout layers are added.
